core/store/src/trie/mod.rs

```diff
@@ -261,9 +261,9 @@ enum RawTrieNode {
 /// Trie node + memory cost of its subtree
 /// memory_usage is serialized, stored, and contributes to hash
 #[derive(Debug, Eq, PartialEq)]
-struct RawTrieNodeWithSize {
-    node: RawTrieNode,
-    memory_usage: u64,
+pub struct RawTrieNodeWithSize {
+    pub node: RawTrieNode,
+    pub memory_usage: u64,
 }
```

Comment:
It's unfortunate that we have to make a bunch of stuff public, but I don't really see a way around this. 

Though, *maaaybe* if we can add something a touch more encapsulated here?

like 

```rust
pub mod estimator {
   pub fn encode_extension_node(key: &[u8]) -> Vec<u8>
}
```

The problem with making stuff `pub` is that it's not clear that it is pub for this one very specific use-case, rather "generally pub".

Reply:
I agree. And I like your suggestion to, let me implement that.

Reply:
Done, now the types and the "real" encoding / decoding functions are back to being private, as they should be.